proselint/checks/__init__.py

```diff
@@ -1 +1,394 @@
+                    start = None
+                    s = 0
+                else:
+                    s = 1
+            pc = c
+        return ranges
+
+    def position_in_ranges(ranges: list[tuple[int, int]], _position: int) -> bool:
+        return any(start <= _position < end for start, end in ranges)
+
+    return position_in_ranges(find_ranges(text), position)
+
+
+###############################################################################
+# The actual check-sub-functions used by the checks  ##########################
+###############################################################################
+
+
+def consistency_check(
+    text: str,
+    word_pairs: list,
+    err: str,
+    msg: str,
+    offset: int = 0,
+) -> list[ResultCheck]:
+    """Build a consistency checker for the given word_pairs."""
+    results = []
+
+    for w in word_pairs:
+        matches = [
+            list(re.finditer(w[0], text)),
+            list(re.finditer(w[1], text)),
+        ]
+
+        if len(matches[0]) > 0 and len(matches[1]) > 0:
+            idx_minority = len(matches[0]) > len(matches[1])
+
+            results += [
+                (
+                    m.start() + offset,
+                    m.end() + offset,
+                    err,
+                    msg.format(w[not idx_minority], m.group(0)),
+                    w[not idx_minority],
+                )
+                for m in matches[idx_minority]
+            ]
+
+    return results
+
+
+def preferred_forms_check(  # noqa: PLR0913, PLR0917
+    text: str,
+    items: list,
+    err: str,
+    msg: str,
+    ignore_case: bool = True,
+    offset: int = 0,
+) -> list[ResultCheck]:
+    """Build a checker that suggests the preferred form."""
+    flags = re.IGNORECASE if ignore_case else 0
+    regex = r"[\W^]{}[\W$]"  # TODO: replace with enum below?
+
+    return [
+        (
+            m.start() + 1 + offset,
+            m.end() + offset,
+            err,
+            msg.format(item[0], m.group(0).strip()),
+            item[0],
+        )
+        for item in items
+        for r in item[1]
+        for m in re.finditer(regex.format(r), text, flags=flags)
+    ]
+    # TODO: can we speed up str.format() ?
+    #       fast-string? or do padding already in checks
+
+
+# PADDINGS, add more if needed
+class Pd(str, Enum):
+    disabled = r"{}"
+    # choose for checks with custom regex
+    whitespace = r"\s{}\s"
+    # -> req whitespace around (no punctuation!)
+    sep_in_txt = r"(?:^|\W){}[\W$]"
+    # req non-text character around
+    # -> finds item as long it is surrounded by any non-word character:
+    #       - whitespace
+    #       - punctuation
+    #       - newline ...
+
+
+def existence_check(  # noqa: PLR0913, PLR0917
+    text: str,
+    re_items: list,
+    err: str,
+    msg: str,
+    ignore_case: bool = True,
+    string: bool = False,  # todo: why not default on?
+    offset: int = 0,  # todo: some checks set this strangely
```

Comment:
```suggestion
    string: bool = False,  # TODO: why not default on?
    offset: int = 0,  # TODO: some checks set this strangely
```
Both interesting points. Which checks set this "strangely", so I can investigate?

Reply:
already resolved. offset-usage is fixed in latest /dev. now only one check is using custom `offset=` as it needs its own padding.